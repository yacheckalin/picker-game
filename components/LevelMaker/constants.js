export const MIN_MAP_SIZE = 3;
export const MAX_MAP_SIZE = 50;
export const DEFAULT_GRID_MAP = [
  [0, 0, 0],
  [0, 0, 0],
  [0, 0, 0],
];

export const DEFAULT_CELL_SIZE = 40;
export const MARK_SELECTED = 999;
