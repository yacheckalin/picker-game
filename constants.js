export const KEYS = {
  BLUE_KEY: [20, "BLUE"],
  GREEN_KEY: [21, "GREEN"],
  RED_KEY: [22, "RED"],
};
export const DOORS = {
  BLUE_DOOR: [90, "BLUE"],
  GREEN_DOOR: [91, "GREEN"],
  RED_DOOR: [92, "RED"],
};
export const WALL = 1;
export const WALL_D = 2; // walls behind the door
export const VISITED = 8;
export const EMPTY = 0;
export const EMPTY_BACKPACK_CELL = 0;
export const DEFAULT_FIELD_SIZE = 4;
export const DEFAUTL_FIELD_HASH = "level_default";

export const MIN_MAP_SIZE = 3;
export const MAX_MAP_SIZE = 50;
export const DEFAULT_GRID_MAP = [
  [0, 0, 0],
  [0, 0, 0],
  [0, 0, 0],
];

export const DEFAULT_CELL_SIZE = 40;
export const MARK_SELECTED = 999;
export const MARK_CLEAR = 0;
