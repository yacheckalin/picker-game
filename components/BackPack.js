import React from "react";
import styled from "styled-components";
import PropTypes from 'prop-types';

import { KEYS } from "../constants";
import { stuffColorPicker, isElementKey } from "../helpers";

const StyledCell = styled.td`
  width: ${(prop) => prop.width}px;
  height: ${(prop) => prop.height}px;
  margin: 0px;
  background-color: ${(prop) => prop.color};
`;

const BackPack = ({ data, cellSize }) => {
  return (
    <table>
      <tbody>
        {data.map((el, index) => (
          <tr key={index}>
            <StyledCell
              width={cellSize}
              height={cellSize}
              className="green center stack-key"
            >
              {isElementKey(el) && (
                <i
                  className={`material-icons  text-darken-4 ${stuffColorPicker(
                    el
                  )}-text`}
                >
                  vpn_key
                </i>
              )}
            </StyledCell>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

BackPack.propTypes = {
  data: PropTypes.array,
  cellSize: PropTypes.number.isRequired
}

export default BackPack;
