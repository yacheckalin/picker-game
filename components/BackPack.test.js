import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

import BackPack from "./BackPack";
import { KEYS } from "../constants";

describe("<BackPack />", () => {
  const props = {
    data: [0, 0, 0, KEYS.GREEN_KEY, 0, 0, KEYS.BLUE_KEY],
    cellSize: 20,
  };

  test("renders and displays properly", () => {
    const wrapper = mount(<BackPack {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("render all keys", () => {
    const wrapper = mount(<BackPack {...props} />);
    expect(wrapper.find(".text-green")).toBeTruthy();
    expect(wrapper.find(".text-blue")).toBeTruthy();
  });

  test("renders backpack with the right size", () => {
    const wrapper = mount(<BackPack {...props} />);
    expect(wrapper.find("tr")).toHaveLength(props.data.length);
  });

  test("set the right height and weight from the cellSize property", () => {
    const wrapper = mount(<BackPack {...props} />);
    expect(wrapper.find(".stack-key").first().prop("width")).toBe(
      props.cellSize
    );
  });
});
