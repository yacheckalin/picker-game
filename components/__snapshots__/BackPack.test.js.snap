// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackPack /> renders and displays properly 1`] = `
<BackPack
  cellSize={20}
  data={
    Array [
      0,
      0,
      0,
      Array [
        21,
        "GREEN",
      ],
      0,
      0,
      Array [
        20,
        "BLUE",
      ],
    ]
  }
>
  <table>
    <tbody>
      <tr
        key="0"
      >
        <styled.td
          className="green center stack-key"
          height={20}
          width={20}
        >
          <td
            className="sc-AxirZ dhaQcU green center stack-key"
            height={20}
            width={20}
          />
        </styled.td>
      </tr>
      <tr
        key="1"
      >
        <styled.td
          className="green center stack-key"
          height={20}
          width={20}
        >
          <td
            className="sc-AxirZ dhaQcU green center stack-key"
            height={20}
            width={20}
          />
        </styled.td>
      </tr>
      <tr
        key="2"
      >
        <styled.td
          className="green center stack-key"
          height={20}
          width={20}
        >
          <td
            className="sc-AxirZ dhaQcU green center stack-key"
            height={20}
            width={20}
          />
        </styled.td>
      </tr>
      <tr
        key="3"
      >
        <styled.td
          className="green center stack-key"
          height={20}
          width={20}
        >
          <td
            className="sc-AxirZ dhaQcU green center stack-key"
            height={20}
            width={20}
          >
            <LMGridObjectIcon
              className=""
              tag={
                Array [
                  21,
                  "GREEN",
                ]
              }
            >
              <styled.i
                className="material-icons text-darken-2 green-text  "
              >
                <i
                  className="sc-AxjAm ipUlGN material-icons text-darken-2 green-text  "
                >
                  vpn_key
                </i>
              </styled.i>
            </LMGridObjectIcon>
          </td>
        </styled.td>
      </tr>
      <tr
        key="4"
      >
        <styled.td
          className="green center stack-key"
          height={20}
          width={20}
        >
          <td
            className="sc-AxirZ dhaQcU green center stack-key"
            height={20}
            width={20}
          />
        </styled.td>
      </tr>
      <tr
        key="5"
      >
        <styled.td
          className="green center stack-key"
          height={20}
          width={20}
        >
          <td
            className="sc-AxirZ dhaQcU green center stack-key"
            height={20}
            width={20}
          />
        </styled.td>
      </tr>
      <tr
        key="6"
      >
        <styled.td
          className="green center stack-key"
          height={20}
          width={20}
        >
          <td
            className="sc-AxirZ dhaQcU green center stack-key"
            height={20}
            width={20}
          >
            <LMGridObjectIcon
              className=""
              tag={
                Array [
                  20,
                  "BLUE",
                ]
              }
            >
              <styled.i
                className="material-icons text-darken-2 blue-text  "
              >
                <i
                  className="sc-AxjAm ipUlGN material-icons text-darken-2 blue-text  "
                >
                  vpn_key
                </i>
              </styled.i>
            </LMGridObjectIcon>
          </td>
        </styled.td>
      </tr>
    </tbody>
  </table>
</BackPack>
`;
